### DISCUSSION TITLE
Download the ESM-1b model onto a local disk

### DISCUSSION THREAD

**Discussion by dongli96** [facebookresearch/esm#120] [Q&A] [answered]
Hi,

With `model, alphabet = torch.hub.load("facebookresearch/esm", "esm1b_t33_650M_UR50S")`  it's a bit annoying to download the large 7GB file _esm1b_t33_650M_UR50S.pt_ every time (even "using cache in /root/.cache/torch/hub/..." costs a lot of time).  

I downloaded the model onto my local disk. How could I use the local model?

Thanks for your time!

**Comment by oliverdutton** [comment, marked as answer]
There's a method for local loading.

```python
model, alphabet = esm.pretrained.load_model_and_alphabet_local(path)
```
